server/configuration.go

```diff
@@ -85,6 +85,15 @@ func (p *Plugin) setConfiguration(configuration *configuration) {
 
 // OnConfigurationChange is invoked when configuration changes may have been made.
 func (p *Plugin) OnConfigurationChange() error {
+	serverConfig := p.API.GetConfig()
```

Comment:
I think this can deadlock. If you move initTelemetry down below where EnsureBots is we are guaranteed to have the config.
You also have to stick it in OnActivate.

Reply:
This was replicated from calls code. And I think the OnConfigurationChange is called at least one on first load, so it should be fine in that sense.